wdl-lint/src/rules/expression_spacing.rs

```diff
@@ -0,0 +1,724 @@
+/// constructs
+fn multiline_else_space(span: Span) -> Diagnostic {
+    Diagnostic::note("multi-line if...then...else must have a preceding space")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a newline before the else keyword")
+}
+
+/// Reports missing close paren for multiline if...then...else constructs
+fn multiline_if_close_paren(span: Span) -> Diagnostic {
+    Diagnostic::note("multi-line if...then...else must have a following newline and parenthesis")
+        .with_rule(ID)
+        .with_highlight(span)
+        .with_fix("add a newline and close parenthesis after to this else clause")
+}
+
+/// Reports missing newline following open brace/bracket/paren for multiline
+/// array/map/object literals
+fn multiline_literal_open_newline(span: Span) -> Diagnostic {
+    Diagnostic::note(
+        "multi-line array/map/object literals must have a newline following the opening token",
+    )
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix("add a newline after this token")
+}
+
+/// Reports missing newline before close brace/bracket/paren for multiline
+/// array/map/object literals
+fn multiline_literal_close_newline(span: Span) -> Diagnostic {
+    Diagnostic::note(
+        "multi-line array/map/object literals must have a newline preceding the closing token",
+    )
+    .with_rule(ID)
+    .with_highlight(span)
+    .with_fix("add a newline before this token")
+}
+
+/// Detects improperly spaced expressions.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct ExpressionSpacingRule;
+
+impl Rule for ExpressionSpacingRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that WDL expressions are properly spaced."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Proper spacing is important for readability and consistency. This rule ensures that \
+         expressions are spaced properly.
+         
+         The following tokens should be surrounded by whitespace when used as an infix: `=`, `==`, \
+         `!=`, `&&`, `||`, `<`, `<=`, `>`, `>=`, `+`, `-`, `*`, `/`, and `%`.
+         
+         The following tokens should not be followed by whitespace when used as a prefix: `-`, and \
+         `!`.
+         
+         Opening brackets (`(`, `[`, and `{`) should not be followed by a space, but may be \
+         followed by a newline. Closing brackets (`)`, `]`, and `}`) should not be preceded by a \
+         space, but may be preceded by a newline.
+         
+         Sometimes a long expression will exceed the maximum line width. In these cases, one or \
+         more linebreaks must be introduced. Line continuations should be indented one more level \
+         than the beginning of the expression. There should never be more than one level of \
+         indentation change per-line. If bracketed content (things between `()`, `[]`, or `{}`) \
```

Comment:
```suggestion
         indentation change per-line.
         
         If bracketed content (things between `()`, `[]`, or `{}`) \
```

Reply:
I wondered about putting a newline there. I've added it now.